### PR TITLE
Using HTML based UI for Change method parameters refactoring

### PR DESCRIPTION
**Pull request by JaroslavTulach**
This PR builds on top of #3357 and 9e27e191c829978a9dbdcdaeee774d4208412348 (accidentally integrated into `master`) to deliver the final vision of #3349 - e.g. to display _rich HTML user interface_ for change method parameters refactoring. Most of the changes (like https://github.com/apache/netbeans/pull/3349/commits/54ddd6ffacd6709c06b793981eafa7b6c9fdc8ab) were done by Dušan. I might have polished them a bit, however my main task is to squash them into a PR with simpler change history.

obrazek

Should there be any problems with 9e27e191c829978a9dbdcdaeee774d4208412348 I am ready to resolve it as part of this PR as soon as possible.

### CODE REVIEW

File: java/java.lsp.server/licenseinfo.xml [hunk 1]
```diff
@@ -27,6 +27,11 @@
         <file>vscode/src/extension.ts</file>
         <license ref="MIT-vscode-ext" />
     </fileset>
+    <fileset>
+        <file>src/org/netbeans/modules/java/lsp/server/refactoring/ui/codicon.ttf</file>
+        <file>src/org/netbeans/modules/java/lsp/server/refactoring/ui/codicon.css</file>
```
Comment (JaroslavTulach) [inline java/java.lsp.server/licenseinfo.xml:32]: These files are taken from [Microsoft codicons](https://github.com/microsoft/vscode-codicons) project. Readme mentions it is MIT licensed and points to [standard MIT license](https://opensource.org/licenses/MIT).

File: java/java.lsp.server/nbproject/org-netbeans-modules-java-lsp-server.sig [hunk 2]
```diff
@@ -23,6 +23,35 @@ meth public void notifyLater(org.openide.NotifyDescriptor)
...
+meth public java.lang.Object createButton(org.netbeans.modules.java.lsp.server.ui.AbstractLspHtmlViewer$View,java.lang.String)
+meth public java.lang.String getId(org.netbeans.modules.java.lsp.server.ui.AbstractLspHtmlViewer$View,java.lang.Object)
+meth public org.netbeans.modules.java.lsp.server.ui.AbstractLspHtmlViewer$View newView(org.netbeans.spi.htmlui.HTMLViewerSpi$Context)
+meth public void runLater(org.netbeans.modules.java.lsp.server.ui.AbstractLspHtmlViewer$View,java.lang.Runnable)
+meth public void setEnabled(org.netbeans.modules.java.lsp.server.ui.AbstractLspHtmlViewer$View,java.lang.Object,boolean)
+meth public void setText(org.netbeans.modules.java.lsp.server.ui.AbstractLspHtmlViewer$View,java.lang.Object,java.lang.String)
+supr java.lang.Object
+hfds initial
+
+CLSS protected final org.netbeans.modules.java.lsp.server.ui.AbstractLspHtmlViewer$View
```
Comment (JaroslavTulach) [inline java/java.lsp.server/nbproject/org-netbeans-modules-java-lsp-server.sig:50]: sigtest doesn't like private `View` - making it `protected` to fix `ant gen-sigtests-release` problem introduced in 9e27e191c829978a9dbdcdaeee774d4208412348

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/htmlui/Buttons.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            + "  footer = document.createElement('div');\n"
+            + "  footer.id = 'dialog-buttons';\n"
+            + "  footer.classList.add('flex');\n"
+            + "  footer.classList.add('section');\n"
+            + "  document.body.appendChild(footer);\n"
+            + "}\n"
+            + "var button = document.createElement('button');\n"
+            + "button.id = id;\n"
+            + "button.onclick = function() {;\n"
+            + "  @org.netbeans.modules.java.lsp.server.htmlui.Buttons::clickButton0(Ljava/lang/String;Ljava/lang/Object;)(id, callback);\n"
```
Comment (JaroslavTulach) [inline java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/htmlui/Buttons.java:55]: The HTML/Java API infrastructure generates a callback class `$JsCallbacks$` in the same package to be able to call from JavaScript to JVM. We don't want that class in the API - as such I am introducing new `Buttons` class and moving all `@JavaScriptBody` methods in here.

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/ui/AbstractLspHtmlViewer.java [hunk 4]
```diff
@@ -44,51 +52,12 @@ public View newView(Context context) {
...
-            + "  footer = document.createElement('div');\n"
-            + "  footer.id = 'dialog-buttons';\n"
-            + "  footer.classList.add('flex');\n"
-            + "  footer.classList.add('section');\n"
-            + "  document.body.appendChild(footer);\n"
-            + "}\n"
-            + "var button = document.createElement('button');\n"
-            + "button.id = id;\n"
-            + "button.onclick = function() {;\n"
-            + "  @org.netbeans.modules.java.lsp.server.ui.AbstractLspHtmlViewer::clickButton0(Ljava/lang/String;Ljava/lang/Object;)(id, callback);\n"
```
Comment (JaroslavTulach) [inline java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/ui/AbstractLspHtmlViewer.java:76]: Moved into `Buttons` class in non-API package.

File: java/java.lsp.server/test/unit/src/org/netbeans/modules/java/lsp/server/ui/MockHtmlViewer.java [hunk 5]
```diff
@@ -74,6 +73,15 @@ private void load(View view, ClassLoader loader, URL pageUrl, Callable<Object> i
         return clazz.cast(v);
     }
 
+    @Override
+    public <C> C component(View view, Class<C> type) {
+        if (type == Void.class) {
+            load(view);
```
Comment (JaroslavTulach) [inline java/java.lsp.server/test/unit/src/org/netbeans/modules/java/lsp/server/ui/MockHtmlViewer.java:72]: Thanks, @dbalek , loading the view after `newView` is finished was one of recent changes. Previously the initialization started sooner then reference to `View` was returned and that lead to various races. Sometimes it worked, sometimes it deadlocked depending on the order of threads execution. Creating the instance first and initializing later when `component(Void.class)` is called fixes these races. I have probably forgotten to update the test - but it actually passed fine this morning (probably too positive order of execution).

**Review by sdedic — APPROVED** [review]
Looks sane, seems to work (change method parameters refactoring)

**Pull request merged**